zebrad/src/commands/copy_state.rs

```diff
@@ -0,0 +1,437 @@
+                      the source state uses the main zebrad config")]
+    target_config_path: Option<PathBuf>,
+
+    /// Filter strings which override the config file and defaults
+    #[options(free, help = "tracing filters which override the zebrad.toml config")]
+    filters: Vec<String>,
+}
+
+impl CopyStateCmd {
+    /// Configure and launch the copy command
+    async fn start(&self) -> Result<(), Report> {
+        let base_config = app_config().clone();
+        let source_config = base_config.state.clone();
+
+        // The default load_config impl doesn't actually modify the app config.
+        let target_config = self
+            .target_config_path
+            .as_ref()
+            .map(|path| app_writer().load_config(path))
+            .transpose()?
+            .map(|app_config| app_config.state)
+            .unwrap_or_else(new_zs::Config::ephemeral);
+
+        info!(?base_config, "state copy base config");
+
+        self.copy(base_config.network.network, source_config, target_config)
+            .await
+            .map_err(|e| eyre!(e))
+    }
+
+    /// Initialize the source and target states,
+    /// then copy from the source to the target state.
+    async fn copy(
+        &self,
+        network: Network,
+        source_config: old_zs::Config,
+        target_config: new_zs::Config,
+    ) -> Result<(), BoxError> {
+        info!(
+            ?source_config,
+            "initializing source state service (old format)"
+        );
+
+        let source_start_time = Instant::now();
+        let (mut source_state, _source_latest_chain_tip, _source_chain_tip_change) =
+            old_zs::init(source_config.clone(), network);
+
+        let elapsed = source_start_time.elapsed();
+        info!(?elapsed, "finished initializing source state service");
+
+        info!(
+            ?target_config, target_config_path = ?self.target_config_path,
+            "initializing target state service (new format)"
+        );
+
+        let target_start_time = Instant::now();
+        let (mut target_state, _target_latest_chain_tip, _target_chain_tip_change) =
+            new_zs::init(target_config.clone(), network);
+
+        let elapsed = target_start_time.elapsed();
+        info!(?elapsed, "finished initializing target state service");
+
+        info!("fetching source and target tip heights");
+
+        let source_tip = source_state
+            .ready()
+            .await?
+            .call(old_zs::Request::Tip)
+            .await?;
+        let source_tip = match source_tip {
+            old_zs::Response::Tip(Some(source_tip)) => source_tip,
+            old_zs::Response::Tip(None) => Err("empty source state: no blocks to copy")?,
+
+            response => Err(format!(
+                "unexpected response to Tip request: {:?}",
+                response,
+            ))?,
+        };
+        let source_tip_height = source_tip.0 .0;
```

Comment:
This is odd, is this a GitHub render artifact?

![image](https://user-images.githubusercontent.com/552961/147014218-28ad155c-0f75-4fc4-8290-37a0d21729f0.png)

Reply:
It's actually a `rustfmt` feature, to work around a `rustc` bug. The `rustc` bug was fixed in 2020, and the `rustfmt` rules were changed in version 2.

I opened ticket #3280 to fix this.